login/controllers/usersController.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const User = require("../models/user");
+const passport = require("passport");
+const getUserParams = body => {
+  return {
+    username: body.username,
+    email: body.email,
+    password: body.password
+  };
+};
+
+module.exports = {
+  index: (req, res, next) => {
+    User.find()
+      .then(users => {
+        res.locals.users = users;
+        next();
+      })
+      .catch(error => {
+        console.log(`Error fetching users: ${error.message}`);
+        next(error);
+      });
+  },
+  indexView: (req, res) => {
+    res.render("users/index");
+  },
+
+  new: (req, res) => {
+    res.render("users/new");
+  },
+  create: (req, res, next) => {
+    if (req.skip) return next();
+    let newUser = new User(getUserParams(req.body));
+    User.register(newUser, req.body.password, (e, user) => {
+      if (user) {
+        req.flash(
+          "success",
+          `${user.username}'s account created successfully!`
+        );
+        res.locals.redirect = "/users";
+        next();
+      } else {
+        req.flash(
+          "error",
+          `Failed to create user account because: ${e.message}`
+          );
+        res.locals.redirect = "/users/new";
+        next();
+      }
+    });
+  },
+  redirectView: (req, res, next) => {
+    let redirectPath = res.locals.redirect;
+    if (redirectPath !== undefined) res.redirect(redirectPath);
```

Comment:
以下のように条件を記述できます。
```suggestion
    if (redirectPath) res.redirect(redirectPath);
```
`===  undefined`のようにすることはなく、falsyな値かどうかで条件を分岐させることが多いです。

ちなみに、
```
if(!redirectPath){
　//異常系だった場合の処理
　return
}
//正常系の処理
```
のように、if文は異常系の処理を先に記述する方が可読性が高いです。

Reply:
考えられていなかったのでご指摘ありがとうございました！
今後に生かします！